modules/tracker/controllers/TrendController.php

```diff
@@ -524,4 +524,46 @@ public function notifysubmitAction()
 
         echo JsonComponent::encode(array('status' => 'ok', 'message' => 'Changes saved'));
     }
+
+    /**
+     * Change key metric status of a trend.
+     *
+     * Request parameters:
+     *     trendId - The id of the producer to delete
+     *     state   - The state of is_key_metric on the trend
+     *
+     * @throws Zend_Exception
+     */
+    public function setkeymetricAction()
+    {
+        $this->disableLayout();
+        $this->disableView();
+
+        /** @var int $trendId */
+        $trendId = $this->getParam('trendId');
+
+        /** @var int $state */
+        $state = $this->getParam('state');
+
+        if (!isset($trendId)) {
+            throw new Zend_Exception('The required trendId parameter is missing.');
+        }
+        if (!isset($state)) {
+            throw new Zend_Exception('The required state parameter is missing.');
+        }
+
+        /** @var Tracker_TrendDao $trendDao */
+        $trendDao = $this->Tracker_Trend->load($trendId);
+
+        /** @var Tracker_ProducerDao $producerDao */
+        $producerDao = $trendDao->getProducer();
+
+        if ($this->Tracker_Producer->policyCheck($producerDao, $this->userSession->Dao, MIDAS_POLICY_ADMIN) === false
+        ) {
+            throw new Zend_Exception('The producer does not exist or you do not have the necessary permission on its community', 403);
+        }
+
+        $trendDao->setIsKeyMetric($state === 'true');
```

Comment:
`1` would be preferable as the output of the key metric field generated by an API call would be `0` or `1`.